lockfile_generator/src/pip.rs

```diff
@@ -0,0 +1,126 @@
+//! Python pip ecosystem.
+
+use std::fmt::Write;
+use std::fs;
+use std::path::Path;
+use std::process::{Command, Stdio};
+
+use serde::Deserialize;
+
+use crate::{Error, Generator, Result};
+
+pub struct Pip;
+
+impl Generator for Pip {
+    fn lockfile_name(&self) -> &'static str {
+        // NOTE: Pip's `generate_lockfile` will never write to disk.
+        unreachable!()
+    }
+
+    fn command(&self, manifest_path: &Path) -> Command {
+        let mut command = Command::new("pip3");
+        command.args(["install", "--quiet", "--report", "-", "--dry-run"]);
```

Comment:
We also want `--ignore-installed`, right? The goal here is to resolve the entire dependency tree.

Reply:
Ah I only ever tried it without anything installed, so that's probably a good point.